### PR TITLE
PR_for_BTCtracker_golang

### PR DESCRIPTION
**Pull request by manas221**
Description
Golang script to fetch the current price of bitcoin in INR using the gocolly package.

Fixes [issue#789](https://github.com/HarshCasper/Rotten-Scripts/issues/789)

## Type of change

_Please delete options that are not relevant._

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Go/Bitcoin_Price_Tracker/Readme.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Bitcoin_Price_Tracker
+This is a script written in **golang** to fetch the live price of **1 BTC** in **INR**. I have used the [colly package](https://github.com/gocolly/colly) to scrape the
+webpage and search the relevant information.
+
```
Comment (HarshCasper) [inline Go/Bitcoin_Price_Tracker/Readme.md:4]: Can you please follow this [template](https://github.com/HarshCasper/Rotten-Scripts/blob/master/TEMPLATE_README.md) ?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by manas221** [comment]
Made the changes on readme as asked according to the template, ignored sections which were not relevant.

File: Go/Bitcoin_Price_Tracker/Readme.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+## Detailed explanation of script
+
+The code uses **colly package** for scraping and **fmt package** for display.
+The *main()* function calls the *give_current_price_of_bitcoin()* which calls an instance of colly's collector to scrape with a boolean based condition to exit out of the target
+search.
+
+## Output
+
+![](output.gif)
```
Comment (HarshCasper) [inline Go/Bitcoin_Price_Tracker/Readme.md:24]: Can you upload the GIF on Imgur and reference it here? Also remove the GIF file from the commits directly

File: Go/Bitcoin_Price_Tracker/Readme.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Bitcoin_Price_Tracker
+
+A script used to fetch the live price of 1 BTC in INR, written in GOlang. [Colly package](https://github.com/gocolly/colly) is used as a base for the web scraping and tag search.
+
+- Fetches the live price of 1 Bitcoin.
+
+## Setup instructions
+
+After forking the repository and cloning on user system ,follow these commands on git bash/bash:
```
Comment (HarshCasper) [inline Go/Bitcoin_Price_Tracker/Readme.md:9]: Keep it `After forking the repository on your local machine, follow these commands on your terminal:`

File: Go/Bitcoin_Price_Tracker/Readme.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Bitcoin_Price_Tracker
+
+A script used to fetch the live price of 1 BTC in INR, written in GOlang. [Colly package](https://github.com/gocolly/colly) is used as a base for the web scraping and tag search.
+
+- Fetches the live price of 1 Bitcoin.
```
Comment (HarshCasper) [inline Go/Bitcoin_Price_Tracker/Readme.md:5]: You can remove this maybe. The information has already been repeated above.

File: Go/Bitcoin_Price_Tracker/main.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+func main() {
+	give_current_price_of_bitcoin()
+}
+
+func give_current_price_of_bitcoin() {
+	c := colly.NewCollector(
+		colly.AllowedDomains("google.com", "www.google.com"),
+	)
+
+	found := false //bool variable to close the search after target tag is found
```
Comment (HarshCasper) [inline Go/Bitcoin_Price_Tracker/main.go:18]: Use comments above the line of code not beside it

File: Go/Bitcoin_Price_Tracker/main.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+		if found {
+			return //return nothing when target tag is matched
+		}
+
+		if e.Attr("class") == "BNeawe iBp4i AP7Wnd" {
+			fmt.Printf("1 BTC = %s\n", e.Text)
+			found = true
+		}
+	})
+	c.Visit("https://www.google.com/search?q=bitcoin+price") //webpage url to be visited
```
Comment (HarshCasper) [inline Go/Bitcoin_Price_Tracker/main.go:31]: Use comments above the line of code not beside it

File: Go/Bitcoin_Price_Tracker/main.go [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+	c := colly.NewCollector(
+		colly.AllowedDomains("google.com", "www.google.com"),
+	)
+
+	found := false //bool variable to close the search after target tag is found
+
+	c.OnHTML("div", func(e *colly.HTMLElement) {
+
+		if found {
+			return //return nothing when target tag is matched
```
Comment (HarshCasper) [inline Go/Bitcoin_Price_Tracker/main.go:23]: Redundant Comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Go/Bitcoin_Price_Tracker/Readme.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Bitcoin_Price_Tracker
+
+A script used to fetch the live price of 1 BTC in INR, written in GOlang. [Colly package](https://github.com/gocolly/colly) is used as a base for the web scraping and tag search.
+
+
+## Setup instructions
+
+After forking the repository on your local machine, follow these commands on your terminal:
+```
```
Comment (vybhav72954) [inline Go/Bitcoin_Price_Tracker/Readme.md:9]: Can we have code fence attributes here pls

    ```shell
    commands
    ```

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: Go/Bitcoin_Price_Tracker/Readme.md (hunk 8 above)
Comment (manas221) [reply Go/Bitcoin_Price_Tracker/Readme.md:9]: I am sorry ,but I can you describe in little more detail what I need to change?
Do I write the three commands, just after the three backticks (in that case, the immediate command disappears) or do i need to write "shell commands" before the actual commands?

Comment (vybhav72954) [reply Go/Bitcoin_Price_Tracker/Readme.md:9]: I think [this](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks) will help you

Comment (manas221) [reply Go/Bitcoin_Price_Tracker/Readme.md:9]: Ohhh, my bad, I over thought the issue..

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**